instrumentation_sql.go

```diff
@@ -289,7 +292,14 @@ type DbConnDetails struct {
 	Error        error
 }
 
-func ParseDBConnDetails(connStr string) DbConnDetails {
+func ParseDBConnDetails(connStr, driverName string) DbConnDetails {
+
+	if isDB2driver(driverName) {
+		if details, ok := parseDB2ConnDetailsKV(connStr); ok {
```

Comment:
Please handle the parsing failure case also

Reply:
Good catch! Done